docs/rst/recording/tutorials/basic_example.rst

```diff
@@ -2,9 +2,263 @@
+
+.. literalinclude:: ../../../../resources/dds/TypeLookupService/TypeLookupServicePublisher.cpp
+    :language: C++
+    :lines: 256-271
+
+Then are initialized the Publisher, DDS Topic and DDS DataWriter.
+
+To make the publication, the public member function ``publish()`` is implemented:
+
+1. It create the variable that will contain the user data, ``dynamic_data_``.
+2. Fill that variable with the function ``fill_helloworld_data_(msg)``, explained below.
+
+.. literalinclude:: ../../../../resources/dds/TypeLookupService/TypeLookupServicePublisher.cpp
+    :language: C++
+    :lines: 229-254
+
+The function ``fill_helloworld_data_()`` returns the data to be sent with the information filled in.
+
+First, it is created the ``Dynamic_ptr`` that will be filled in and return.
+With the use of ``DynamicDataFactory`` we create the data that corresponds to our data type.
+Finally, data variables are assigned, in this case, ``index`` and ``message``.
+
+.. literalinclude:: ../../../../resources/dds/TypeLookupService/TypeLookupServicePublisher.cpp
+    :language: C++
+    :lines: 293-307
+
+**************
+DDS Subscriber
+**************
+
+The DDS Subscriber is acting as a client of types.
+Thus, the subscriber does not need to know the type.
+
+Examining the code
+==================
+
+This section explains the C++ source code of the DDS Publisher, which can also be found `here <https://github.com/eProsima/DDS-Recorder/blob/main/resources/dds/TypeLookupService>`_.
+
+The private data members of the class defines the DDS Topic, DDS Topic type and DynamicType.
+
+.. literalinclude:: ../../../../resources/dds/TypeLookupService/TypeLookupServiceSubscriber.h
+    :language: C++
+    :lines: 105-110
+
+The next lines show the constructor of the ``TypeLookupServiceSubscriber`` class that implements the subscriber setting the topic name as the one configured in the
+publisher side.
+
+.. literalinclude:: ../../../../resources/dds/TypeLookupService/TypeLookupServiceSubscriber.cpp
+    :language: C++
+    :lines: 45-53
+
+Inside the ``TypeLookupServiceSubscriber`` constructor are defined the DomainParticipantQos.
+As the publisher act as a client of types, set the QoS in order to receive this information.
+Set ``use_client`` to ``true`` and ``use_server`` to ``false``.
+
+.. literalinclude:: ../../../../resources/dds/TypeLookupService/TypeLookupServiceSubscriber.cpp
+    :language: C++
+    :lines: 57-61
+
+Then is initialized the Subscriber.
+
+Inside ``on_data_available()`` it is created the ``DynamicData_ptr`` where the samples received will be read.
+
+As in the publisher, it is used ``DynamicDataFactory`` for the creation of the data that corresponds to our data type.
+
+.. literalinclude:: ../../../../resources/dds/TypeLookupService/TypeLookupServiceSubscriber.cpp
+    :language: C++
+    :lines: 143-170
+
+The function ``on_type_information_received()`` detect the topics received and discover it if they have the same as the topic that we are waiting for.
```

Comment:
```suggestion
The function ``on_type_information_received()`` detects if new topic information has been received in order to proceed to register the topic in case it has the same name as the expected one. 
```